scripts/audit-all.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+dirsToAudit=$(find . -type f -not -path "*/node_modules*/*" -not -path '*/scripts/*' -name "package.json" | awk '{print substr($0,0,length()-13);}' | sort)
+filter='| select(.value.severity | contains("high","critical"))'
+format='[\(.value.severity)] \(.value.title) for \(.value.module_name) (#\(.key))'
+if [[ $1 == "--all" ]]; then
+   filter=''
+fi
+
+for i in ${dirsToAudit[@]}; do
+  (cd $i && npm audit --package-lock-only --json | jq -r ".advisories | to_entries[] $filter | \"$format in $i\"")
+done
```

Comment:
This is hard to read, not documented, not portable, and has external dependencies. Our policy is to prefer javascript because all developers are familiar with it and it's cross platform, and has dependency management. A quick glance suggests it'd be reasonably easy to rewrite this in javascript - what do you think?

Reply:
I used `jq` because it is already referenced in `Gruntfile` for the `npm-ci-shared-libs` and `shared-libs-unit` tasks.  So although it is an external dependency, it is one that is already required for existing tasks.  Similarly it is used in README.md and six scritps.

The documentation and readability concerns seem resolvable.  I'm not sure what you mean by not portable.

Reply:
I mean cross-platform portability.